### PR TITLE
ADP-1858-upgrade-from-csl-to-cml

### PR DESCRIPTION
**Pull request by AngelCastilloB**
# Context

The CSL was compiled to WASM using an old version of rustwasm which had a [bug](https://github.com/rustwasm/wasm-bindgen/issues/1963) that would leak stack space when errors were thrown by the underlying rust library, This was making the CSL WASM module to crash after a several errors were thrown, this was fixed in later versions. The CML library uses the updated rustwasm and as such doesnt have this bug.

# Proposed Solution

Reemplace the CSL library with the CML library

### CODE REVIEW

**Review by rhyslbw — CHANGES_REQUESTED** [review]
I had a heart-in-mouth moment :laughing::
image

Did you run `npm i`? Please ensure `yarn install` has been run again after removing the `package-lock.json`

File: packages/input-selection/src/selectionConstraints.ts [hunk 1]
```diff
@@ -28,14 +28,24 @@ export const computeMinimumCost =
   ): EstimateTxFee =>
   async (selection) => {
     const tx = await buildTx(selection);
+
     return usingAutoFree((scope) => {
+      // TODO: Get this from cardano-services
```
Comment (mkazlauskas) [inline packages/input-selection/src/selectionConstraints.ts:33]: Just flagging this to be resolved

File: packages/wallet/test/hardware/LedgerKeyAgent.test.ts [hunk 2]
```diff
@@ -140,7 +140,7 @@ describe('LedgerKeyAgent', () => {
   });
 
   describe('establish, check and re-establish device connection', () => {
-    let deviceConnection: DeviceConnection;
+    let deviceConnection: any;
```
Comment (mkazlauskas) [inline packages/wallet/test/hardware/LedgerKeyAgent.test.ts:143]: Let's try to figure out why this is needed

**Review by mkazlauskas — CHANGES_REQUESTED** [review]

File: packages/e2e/package.json [hunk 3]
```diff
@@ -65,7 +65,7 @@
     "access": "public"
   },
   "dependencies": {
-    "@cardano-foundation/ledgerjs-hw-app-cardano": "5.0.0",
+    "@cardano-foundation/ledgerjs-hw-app-cardano": "^5.0.0",
```
Comment (rhyslbw) [inline packages/e2e/package.json:68]: This should be a separate `chore` commit

File: packages/e2e/src/factories.ts [hunk 4]
```diff
@@ -267,14 +266,12 @@ keyManagementFactory.register('ledger', async (params: any): Promise<CreateKeyAg
...
 
-  let deviceConnection: DeviceConnection | null | undefined;
-
   return async (dependencies) => {
     const ledgerKeyAgent = await LedgerKeyAgent.createWithDevice(
       {
         accountIndex: params.accountIndex,
         communicationType: CommunicationType.Node,
-        deviceConnection,
+        deviceConnection: null,
```
Comment (rhyslbw) [inline packages/e2e/src/factories.ts:274]: Separate `test` commit. Was this a bug?

File: packages/e2e/test/web-extension/wdio.conf.js [hunk 5]
```diff
@@ -50,6 +50,6 @@ module.exports = {
             'chromedriver'
         ],
         specs: ['./**/*.spec.ts'],
-        waitforTimeout: 10000
+        waitforTimeout: 30000
```
Comment (rhyslbw) [inline packages/e2e/test/web-extension/wdio.conf.js:53]: This doesn't belong in the same commit as the CML replacement

**Review by rhyslbw — CHANGES_REQUESTED** [review]
The complete diff looks good, but there's a few distinct changes that have been lumped into a single commit, that need to be separated. I've commented on the ones that stick out to me, but it also appears that we missed including some CSL objects in the managed scope the first time around, so they would be best committed first as `fix`, then the lib replacement can be clean. If the new lib introduces new objects, then that's naturally part of the refactor commit.

File: packages/e2e/test/web-extension/wdio.conf.js (hunk 5 above)
Comment (rhyslbw) [reply packages/e2e/test/web-extension/wdio.conf.js:53]: The commit message is missing some context and includes redundant information already expressed in the diff. Please have another go at this commit message, with an explanation of why this is needed.

**Review by rhyslbw — CHANGES_REQUESTED** [review]
image

Some Spanglish? :smile: 

reemplaced -> replaced

**Comment by rhyslbw** [comment]
> I split the extra manage here: https://github.com/input-output-hk/cardano-js-sdk/commit/b01944c20e258097ad4e08db12e7492e99502482

Great, that's good to see there was only a small number missed in the original implementation

File: packages/input-selection/src/selectionConstraints.ts (hunk 1 above)
Comment (mkazlauskas) [reply packages/input-selection/src/selectionConstraints.ts:33]: Are we planning to do this TODO in a separate PR?

**Review by mkazlauskas — APPROVED** [review]
Nice work!

**Review by rhyslbw — APPROVED** [review]
Great work @AngelCastilloB :construction_worker:

**Review by mkazlauskas — APPROVED** [review]

**Pull request merged**